visualization_marker_tutorials/README.md

```diff
@@ -0,0 +1,265 @@
+
+```
+marker.pose.position.x = 0;
+marker.pose.position.y = 0;
+marker.pose.position.z = 0;
+marker.pose.orientation.x = 0.0;
+marker.pose.orientation.y = 0.0;
+marker.pose.orientation.z = 0.0;
+marker.pose.orientation.w = 1.0;
+```
+Here we set the pose of the marker.
+The `geometry_msgs/msg/Pose` message consists of a `geometry_msgs/msg/Vector3` to specify the position and a `geometry_msgs/msg/Quaternion` to specify the orientation.
+Here we set the position to the origin, and the orientation to the identity orientation (note the 1.0 for w).
+
+```
+marker.scale.x = 1.0;
+marker.scale.y = 1.0;
+marker.scale.z = 1.0;
+```
+Now we specify the scale of the marker.
+For the basic shapes, a scale of 1 in all directions means 1 meter on a side.
+
+```
+marker.color.r = 0.0f;
+marker.color.g = 1.0f;
+marker.color.b = 0.0f;
+marker.color.a = 1.0;
+```
+The color of the marker is specified as a `std_msgs/msg/ColorRGBA`.
+Each member should be between 0 and 1.
+An alpha (a) value of 0 means completely transparent (invisible), and 1 is completely opaque.
+
+```
+marker.lifetime = rclcpp::Duration::from_nanoseconds(0);
+```
+The lifetime field specifies how long this marker should stick around before being automatically deleted. 
+A value of 0 means never to auto-delete.
+If a new marker message is received before the lifetime has been reached, the lifetime will be reset to the value in the new marker message.
+
+```
+while (marker_pub->get_subscription_count() < 1) {
+  if (!rclcpp::ok()) {
+    return 0;
+  }
+  auto logger = rclcpp::get_logger("my_subscriber");
+  RCLCPP_WARN(logger, "Please create a subscriber to the marker");
+  loop_rate.sleep();
+}
+marker_pub->publish(marker);
+```
+We wait for the marker to have a subscriber and then we publish the marker.
+
+```
+switch (shape) {
+  case visualization_msgs::msg::Marker::CUBE:
+    shape = visualization_msgs::msg::Marker::SPHERE;
+    break;
+  case visualization_msgs::msg::Marker::SPHERE:
+    shape = visualization_msgs::msg::Marker::ARROW;
+    break;
+  case visualization_msgs::msg::Marker::ARROW:
+    shape = visualization_msgs::msg::Marker::CYLINDER;
+    break;
+  case visualization_msgs::msg::Marker::CYLINDER:
+    shape = visualization_msgs::msg::Marker::CUBE;
+    break;
+}
+```
+This code lets us show all four shapes while just publishing the one marker message.
+Based on the current shape, we set what the next shape to publish will be.
+
+```
+loop_rate.sleep();
+```
+Sleep and loop back to the top.
+
+### Running the code
+Naviagte into your workspace, then build the code and run it:
+
+```
+$ colcon build --packages-select visualization_marker_tutorials
+$ ros2 run visualization_marker_tutorials basic_shapes
+``` 
+### Viewing the markers
+Now that you're publishing markers, you need to run RViz to view them:
+
+```
+$ rviz2
+```
+
+If you've never used RViz before, please see the User's Guide to get you started.
```

Comment:
Unfortunately, we don't have a ROS 2 equivalent of the user guide. It still might be worth referencing the ROS 1 guide on the wiki and add a TODO here to link to a ROS 2 version when it exists.

Reply:
I've added a link to the ROS 1 guide with a note that the ROS 2 guide is coming in the future.